### DISCUSSION TITLE
How to help people getting started

### DISCUSSION THREAD

**Discussion by roelofz** [nextcloud/all-in-one#4727] [Questions] [answered]
### Is your feature request related to a problem? Please describe. 

The steps to install local-ai (https://github.com/nextcloud/all-in-one/tree/main/community-containers/local-ai) are not complete, The admin manual needs to be followed to maybe get a working configuration:
https://docs.nextcloud.com/server/latest/admin_manual/ai/app_assistant.html#installation
I think this makes a working config, allthough I did not succeed in this (install exapp is getting me lost), but I'm sure the admin guide will help more then not mentioning it here.

### Describe the solution you'd like

Include the admin guide, or include the minimal required steps.

**Comment by roelofz** [comment]
Well, one thing is for sure: I have no idea how to get AI working in AIO. 
Did a lot of searching, but I am affraid the admin guide is a dead end too.

So in general the feature request is: how to configure and use the eye candy.
I'm puzzled how to get it working....

**Comment by szaimen** [comment, marked as answer]
Hi, can you specify a bit more in detail what you are missing in the docs? https://github.com/nextcloud/all-in-one/tree/main/community-containers/local-ai

As for most other AI stuff, there is this container: 
![image](https://github.com/nextcloud/all-in-one/assets/42591237/917e424e-d18b-4b93-ac15-da7afa865047)
that you can enable and then you have an additional point in this dropdown where you can install most other AI apps:
![image](https://github.com/nextcloud/all-in-one/assets/42591237/a4b9ca1e-37e2-46d6-8b84-611b3a894413)

**Reply by roelofz** [reply]
Hi Szaimen,

I have AIO, included the socks proxy in the install, and followed the steps written in https://github.com/nextcloud/all-in-one/tree/main/community-containers/local-ai including the addition of the three models in the example.

At first the local-ai container did not get the required ports on the outside, after removing and re-adding the container, this was working and can be reached from the nextcloud container using the ports in an nc command. The models seem to be loaded too (5 gb in data in the volume folder), but when launching a test in the assistant, an `500 Internal Server Error` shows up in the logs. 

I tried to follow the general admin manual, but this seems not the right way too for the AIO scenario, so I'm puzzled.
There seems to be something missing, allthough I reread the instructions again and again, but I can´t find out what I have been missing.

I looked into Connected accounts, but it is not clear if something needs to be declared over there (only a hint exists regarding chatgpt). 

I finally got some answers!
It looks like I needed to declare GPT4all-j as a text generator in connected accounts and now I'm getting an overheated pc ( :-) ) and answers (this was the missing clue). It may be small but when searching for a first working setup, this is crucial.
Now I too need to look at the admin AI section what model to use for what (at least I think that section needs some attention, as well as discovering maybe other models).
Stable Diffusion is missing a name declaration in the yaml sample, minor but again food for errors I think. 

So I think these are the missing tips, what got me searching quite a while now, but I got some responses, so food for new thoughts.

**Reply by szaimen** [reply]
I see. Would you mind editing this file and submitting a pr? :)
https://github.com/nextcloud/all-in-one/edit/main/community-containers/local-ai/readme.md

**Reply by roelofz** [reply]
Well for everything a first, so I hope it has been committed correctly.
So I was glad to help!
I did some testing, gpt4all-j must be selected, otherwise it will not work.
I tried removing the stable diffusion name (maybe without declaring this it would default back to gpt4all-j), but could not see a clear result of that (it stayed in connected accounts as an option).

**Reply by szaimen** [reply]
Thank you! I just merged your changes in https://github.com/nextcloud/all-in-one/pull/4731 :)
